sqlserver/datadog_checks/sqlserver/database_metrics/database_agent_metrics.py

```diff
@@ -0,0 +1,159 @@
+from datadog_checks.base.config import is_affirmative
+
+from datadog_checks.sqlserver.const import  AWS_RDS_HOSTNAME_SUFFIX
+
+from .base import SqlserverDatabaseMetricsBase
+
+AGENT_ACTIVITY_DURATION_QUERY = {
+    "name": "msdb.dbo.sysjobactivity",
+    "query": """\
+        SELECT
+            sj.name,
+            ja.job_id,
+            DATEDIFF(SECOND, ja.start_execution_date, GETDATE()) AS duration_seconds
+        FROM msdb.dbo.sysjobactivity AS ja
+        INNER JOIN msdb.dbo.sysjobs AS sj
+        ON ja.job_id = sj.job_id
+        WHERE ja.start_execution_date IS NOT NULL
+            AND ja.stop_execution_date IS NULL
+            AND session_id = (
+                SELECT MAX(session_id) 
+                FROM msdb.dbo.sysjobactivity
+            )
+    """,
+    "columns": [
+        {"name": "job_name", "type": "tag"},
+        {"name": "job_id", "type": "tag"},
+        {"name": "agent.active_jobs.duration", "type": "gauge"},
+    ],
+}
+
+AGENT_ACTIVITY_STEPS_QUERY = {
+    "name": "msdb.dbo.sysjobactivity",
+    "query": """\
+        WITH ActiveJobs AS (
+            SELECT
+                ja.job_id,
+                last_executed_step_id
+            FROM msdb.dbo.sysjobactivity AS ja
+            WHERE ja.start_execution_date IS NOT NULL
+                AND ja.stop_execution_date IS NULL
+                AND session_id = (
+                    SELECT MAX(session_id) 
+                    FROM msdb.dbo.sysjobactivity
+                )
+        ),
+        CompletedSteps AS (
+            SELECT
+                sjh1.job_id,
+                sjh1.step_id,
+                sjh1.step_name,
+                sjh1.run_status
+            FROM msdb.dbo.sysjobhistory AS sjh1
+            WHERE NOT EXISTS (
+                SELECT 1
+                FROM msdb.dbo.sysjobhistory AS sjh2
+                WHERE sjh2.job_id = sjh1.job_id
+                    AND sjh2.step_id = 0
+                    AND sjh2.instance_id > sjh1.instance_id
+                )
```

Comment:
I see, it would be possible to select the matching history entry with max instance_id as this would reflect the most recent completion of that step. Would that typically be faster. I can also write up the query if that would be helpful.

Reply:
I would generate ~50000 history entries and measure performance.